.github/workflows/tests.yml

```diff
@@ -37,53 +45,58 @@ jobs:
     - name: Build
       run: |
         echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u ${{ github.actor }} --password-stdin || true
-        docker pull ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}} || true
+        docker pull ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}_${{matrix.kokkos_backend}} || true
         docker build \
-          --cache-from ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}} \
-          -t ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}} \
-          -t ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}:${GITHUB_SHA:0:7} \
-          docker/${{matrix.image}}
+          --cache-from ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}_${{matrix.kokkos_backend}} \
+          -t ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}_${{matrix.kokkos_backend}} \
+          -t ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}_${{matrix.kokkos_backend}}:${GITHUB_SHA:0:7} \
+          docker/${{matrix.image}}_${{matrix.kokkos_backend}}
     - name: Publish image for current SHA
       if: needs.id_repo.outputs.in_base_repo == 'true'
       run: |
-        docker push ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}:${GITHUB_SHA:0:7}
-    - name: Publish latest (default) image 
+        docker push ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}_${{matrix.kokkos_backend}}:${GITHUB_SHA:0:7}
+    - name: Publish latest (default) image
       if: github.event_name == 'push' && github.ref_name == 'main' && needs.id_repo.outputs.in_base_repo == 'true'
       run: |
-        docker push ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}
+        docker push ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}_${{matrix.kokkos_backend}}
     - name: Create image tarball
       if: needs.id_repo.outputs.in_base_repo == 'false'
       run: |
-        docker save ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}:${GITHUB_SHA:0:7} > ${{matrix.image}}.tar
+        docker save ghcr.io/maison-de-la-simulation/ddc/${{matrix.image}}_${{matrix.kokkos_backend}}:${GITHUB_SHA:0:7} > ${{matrix.image}}_${{matrix.kokkos_backend}}.tar
     - name: Generate docker artifact from image
       if: needs.id_repo.outputs.in_base_repo == 'false'
       uses: actions/upload-artifact@v3
       with:
-        name: ${{matrix.image}}-artifact
-        path: ${{matrix.image}}.tar
+        name: ${{matrix.image}}_${{matrix.kokkos_backend}}-artifact
+        path: ${{matrix.image}}_${{matrix.kokkos_backend}}.tar
         retention-days: 1
 
   test:
     strategy:
       fail-fast: false
       matrix:
+        image: ['focal', 'jammy']
+        kokkos_backend: ['CPU', 'CUDA', 'HIP']
         cxx_version:  ['17', '20', '23']
         cmake_build_type: ['Debug', 'Release']
-        kokkos_backend: ['CPU-gnu', 'CPU-clang', 'GPU-HIP', 'GPU-CUDA']
-        image: ['test_env', 'jammy_env']
+        host_compiler: ['GCC', 'Clang']
```

Comment:
shouldn't this only be a choice when kokkos_backend is CPU?

Reply:
Why not but I have difficulties to get the right image depending on the backend and to retrieve the compiler when it is CPU. That is why I introducing a new dimension.

Maybe I could use `include` instead of `exclude` with this new dimension ?